src/components/flat-table/flat-table-head/flat-table-head.style.js

```diff
@@ -4,11 +4,18 @@ import { StyledFlatTableRowHeader } from "../flat-table-row-header/flat-table-ro
 import StyledFlatTableCheckbox from "../flat-table-checkbox/flat-table-checkbox.style";
 
 const StyledFlatTableHead = styled.thead`
-  ${StyledFlatTableRowHeader}, ${StyledFlatTableCheckbox} {
-    border-left: none;
-    border-right: none;
-    font-weight: 700;
-    z-index: ${({ theme }) => theme.zIndex.overlay};
+  &&& {
+    ${StyledFlatTableCheckbox} {
+      border-left: none;
+      border-right: none;
+    }
+    ${StyledFlatTableRowHeader}, ${StyledFlatTableCheckbox} {
+      font-weight: 700;
+    }
+  }
+
+  ${StyledFlatTableRowHeader} {
+    z-index: ${({ theme }) => theme.zIndex.overlay + 5};
```

Comment:
Yeah I was going to do something like this but ended up not as I'm hoping we can removed some of these when we rewrite. I'm happy to add it in as you make a good point and who knows when it will be rewritten 👍